lib/middleware.js

```diff
@@ -31,3 +35,16 @@ module.exports.applyMiddleware = function (argv, middlewares) {
       }
     }, argv)
 }
+
+module.exports.applyPreCheckMiddlware = function (argv, middlewares, context, aliases) {
+  for (let i = 0; i < middlewares.length; i++) {
+    let preChecksContext = {
+      'commands': JSON.parse(JSON.stringify(context.commands)),
```

Comment:
oh, I see what you're doing here, a deep copy so that you can't modify the original object? clever.